index.js

```diff
@@ -91,57 +130,32 @@ async function runScript() {
     data,
   };
 
-  console.log(
-    `PREPARED REQUEST:`,
-    JSON.stringify(omit(request, "headers.Authorization"))
-  );
+  console.log(`PREPARED REQUEST:`, JSON.stringify(omit(request, 'headers.Authorization')));
   console.log(`SENDING...`);
 
   axios
     .request(request)
     .then((response) => {
       console.log(`REQUEST COMPLETED SUCCESSFULLY:`);
-      console.log("  ", response.status, response.statusText);
-      console.log("  ", "Headers:", JSON.stringify(response.headers));
-      console.log("  ", "Body:", JSON.stringify(response.data));
+      console.log('  ', response.status, response.statusText);
+      console.log('  ', 'Headers:', JSON.stringify(response.headers));
+      console.log('  ', 'Body:', JSON.stringify(response.data));
     })
     .catch((error) => {
       if (error.response) {
         // Response received with a non 20x status.
         console.log(`REQUEST RETURNED ERROR:`);
-        console.log("  ", error.response.status, error.response.statusText);
-        console.log("  ", "Headers:", JSON.stringify(error.response.headers));
-        console.log("  ", "Body:", JSON.stringify(error.response.data));
+        console.log('  ', error.response.status, error.response.statusText);
+        console.log('  ', 'Headers:', JSON.stringify(error.response.headers));
+        console.log('  ', 'Body:', JSON.stringify(error.response.data));
       } else {
         console.log(`ERROR SENDING REQUEST:`, error.message);
+
+        process.exitCode = 1;
       }
     });
 }
 
-function consoleSectionBreak() {
-  console.log(`\n\n=========================`);
-}
-
-async function fetchEgoCredentials(egoUrl, egoClient, egoSecret) {
-  consoleSectionBreak();
-  console.log(`FETCHING EGO AUTH AT: ${egoUrl}`);
-  const egoAuthUrl = `${egoUrl}/oauth/token?client_id=${egoClient}&client_secret=${egoSecret}&grant_type=client_credentials`;
-  // https://ego.dev.argo.cancercollaboratory.org/api/oauth/token?grant_type=client_credentials&client_id=daco-cron-runner&client_secret=secret
-
-  try {
-    const response = await axios.post(egoAuthUrl);
-    const egoAuth = response.data.access_token;
-
-    console.log(`EGO CREDENTIALS RETRIEVED, ADDING TO REQUEST`);
-    headers.Authorization = `Bearer ${egoAuth}`;
-    consoleSectionBreak();
-  } catch (err) {
-    console.log(`ERROR WHILE RETRIEVING EGO CREDENTIALS - ${err.message}`);
-    console.log(`ENDING PROCESS DUE TO EGO FAILURE`);
-    process.exit();
```

Comment:
A subroutine shouldn't kill the main process. I've added centralised handling for this in in `runScript`, so that this and any future implementation have a single control point.